From 0b7fb6190f6885a647e9a1fe77c412151fa10968 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 18 Nov 2024 20:55:25 +0100
Subject: [PATCH 5/6] Make seek work on `/dev/null` (#22886)

Seeking `/dev/null` should work and do nothing instead of returning
`EPIPE`. This fixes it.

This can be removed when we update to Emscripten 3.1.72.
---
 src/library_fs.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/library_fs.js b/src/library_fs.js
index b5cf19e5d..5aa03d877 100644
--- a/src/library_fs.js
+++ b/src/library_fs.js
@@ -1355,6 +1355,7 @@ FS.staticInit();
       FS.registerDevice(FS.makedev(1, 3), {
         read: () => 0,
         write: (stream, buffer, offset, length, pos) => length,
+        llseek: () => 0,
       });
       FS.mkdev('/dev/null', FS.makedev(1, 3));
       // setup /dev/tty and /dev/tty1
-- 
2.34.1

